### PR TITLE
Three dimensional image processing tutorial

### PR DESCRIPTION
**Pull request by mcquin**
For SciPy 2017 (& beyond)! @stefanv @emmanuelle 

I'm thinking about adding a bit at the beginning which highlights some of different ways the image spacing can be taken into account. The steps in the tutorial for segmenting the image don't need to considering spacing. Or/also I can add a bit about feature extraction.

Should I provide solutions?

### CODE REVIEW

**Comment by jni** [comment]
Hi @mcquin! Very nice. It seems you are following the SWC convention of not aliasing packages (?), but we try to follow the standard community practices — import numpy as np, from scipy import ndimage as ndi, import matplotlib.pyplot as plt, etc.

I think it would be very nice to consider spacing, since it's quite common in 3D images to have different spacing along the pln axis.

I can't remember what we do with respect to solutions, so I'll let @stefanv handle that one...

**Comment by mcquin** [comment]
Thanks for the feedback @jni ! I don't use import aliases, and I am not familiar with the conventions of the community. I've made your suggested changes, please let me know if there are more. :)

I've also expanded the tutorial to highlight more image processing functions, especially those which are sensitive to spacing. Additionally, I've added a section covering feature extraction. This requires `pandas` and could be removed if adding a new dependency is :-1:'d.

I've added solutions, too!

**Comment by stefanv** [comment]
The installation instructions do not include any mention of Pandas, so I'd push back a bit against that dependency.  I took a look at what you are using it for, and wonder if there's any disadvantage to simply using a dictionary?

Thanks for adding solutions!

**Comment by jni** [comment]
@mcquin Oh! One of those small things that one assumes everyone magically knows for some reason! =) 

```python
import pandas as pd
from skimage import (morphology, segmentation, whatever)
from scipy import (stats, optimize, whatever)
```

@stefanv I don't think it's much of a stretch to think virtually everyone in the tutorial will have pandas installed... For ML applications from image features, it is probably a good thing to have in our dependency list in the tutorials. But I don't feel strongly about this.

**Comment by stefanv** [comment]
Perhaps it's more that I have an allergy for Pandas's complicated API, and I would prefer not to muddle the waters, if a dictionary will suffice.

**Comment by jni** [comment]
@stefanv As you know, I suffer from the same allergy, but I've found that sometimes I just need to take a Claratyne™ and get on with life, because it is in such widespread use. =P But I agree that in this case it doesn't seem to offer much utility.

**Comment by mcquin** [comment]
It seems we have different allergies. ;)

As requested, `pandas` is gone and I've done more import aliasing. I'm flying today, and will respond to any new feedback first thing tomorrow.

File: lectures/three_dimensional_image_processing.ipynb [hunk 1]
```diff
@@ -0,0 +1,864 @@
...
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def get_cmap(labels, name=\"viridis\"):\n",
+    "    cmap = matplotlib.cm.get_cmap(\"viridis\")\n",
+    "    masked_labels = np.ma.masked_where(labels == 0, labels)\n",
```
Comment (mcquin) [inline lectures/three_dimensional_image_processing.ipynb]: This is the trick I know for setting the value 0 to black, regardless of colormap.

File: lectures/three_dimensional_image_processing.ipynb [hunk 2]
```diff
@@ -0,0 +1,864 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def show_plane(ax, plane, cmap=\"gray\", title=None):\n",
```
Comment (mcquin) [inline lectures/three_dimensional_image_processing.ipynb]: `ax` appears consistent with matplotlib. I'll stick with it. :)

**Comment by stefanv** [comment]
Any updates can go straight into master, thanks!

**Pull request merged**